cli/internal/cmd/apply.go

```diff
@@ -419,11 +419,24 @@ func (a *applyCmd) validateInputs(cmd *cobra.Command, configFetcher attestationc
 		return nil, nil, err
 	}
 
+	// Read and validate state file
 	a.log.Debugf("Reading state file from %s", a.flags.pathPrefixer.PrefixPrintablePath(constants.StateFilename))
 	stateFile, err := state.ReadFromFile(a.fileHandler, constants.StateFilename)
 	if err != nil {
 		return nil, nil, err
 	}
+	if a.flags.skipPhases.contains(skipInitPhase) {
+		// If the skipInit flag is set, we are in a state where the cluster
+		// has already been initialized and check against the respective constraints.
+		if err := stateFile.Validate(state.PostInit, conf.GetProvider()); err != nil {
+			return nil, nil, err
+		}
+	} else {
+		// The cluster has not been initialized yet, so we check against the pre-init constraints.
+		if err := stateFile.Validate(state.PreInit, conf.GetProvider()); err != nil {
+			return nil, nil, err
+		}
+	}
```

Comment:
This validation needs to be moved down (to the end of the input validation function), since we don't know what phases are skipped until we parsed all inputs